src/spec.c

```diff
@@ -2085,6 +2092,49 @@ static void Indexes_ScanProc(RedisModuleCtx *ctx, RedisModuleString *keyname, Re
   if (scanner->cancelled) {
     return;
   }
+
+  RedisModuleServerInfoData *memory_data = RedisModule_GetServerInfo(ctx, "memory");
+  const char *error = NULL;
+  if (memory_data) {
+
+    // Get used memory in bytes
+     unsigned long long redis_used_memory = RedisModule_ServerInfoGetFieldUnsigned(memory_data, "used_memory", NULL);
+    // Get max memory in bytes
+    unsigned long long redis_max_memory = RedisModule_ServerInfoGetFieldUnsigned(memory_data, "maxmemory", NULL);
+    // if used memory is more than 80% of max memory, cancel the scan
+    if (redis_max_memory && redis_used_memory > 0.8 * redis_max_memory) {
+      error = "Used memory is more than 80%% of max memory, cancelling the scan";
+      RedisModule_Log(ctx, "warning", error);
+      scanner->cancelled = true;
+    }
+    RedisModule_FreeServerInfo(ctx, memory_data);
+    
+  } else {
+    error = "Failed to get memory info";
+    RedisModule_Log(ctx, "warning", error);
+    scanner->cancelled = true;
+  }
+
+  if(scanner->cancelled) {
+    // We need to report the error message besides the log, so we can show it in FT.INFO
+    if(scanner->global) {
+      // Dvirdu: Do we really get here?
```

Comment:
I think temporary indexes also use the global scanner

Reply:
The only code path I found using global scanner is for legacy index. @MeirShpilraien thoughts?

Reply:
Its not legacy indexes, its when we running on Redis version that does not support the load event. I believe this is no longer relevant right?